contrib/libssh-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+set(LIB_SOURCE_DIR "${ClickHouse_SOURCE_DIR}/contrib/libssh")
+set(LIB_BINARY_DIR "${ClickHouse_BINARY_DIR}/contrib/libssh")
+# Specify search path for CMake modules to be loaded by include()
+# and find_package()
+list(APPEND CMAKE_MODULE_PATH "${LIB_SOURCE_DIR}/cmake/Modules")
+
+include(DefineCMakeDefaults)
+include(DefineCompilerFlags)
+
+project(libssh VERSION 0.9.7 LANGUAGES C)
+
+# global needed variable
+set(APPLICATION_NAME ${PROJECT_NAME})
+
+# SOVERSION scheme: CURRENT.AGE.REVISION
+#   If there was an incompatible interface change:
+#     Increment CURRENT. Set AGE and REVISION to 0
+#   If there was a compatible interface change:
+#     Increment AGE. Set REVISION to 0
+#   If the source code was changed, but there were no interface changes:
+#     Increment REVISION.
+set(LIBRARY_VERSION "4.8.7")
+set(LIBRARY_SOVERSION "4")
+
+# where to look first for cmake modules, before ${CMAKE_ROOT}/Modules/ is checked
+
+# add definitions
+
+include(DefinePlatformDefaults)
+
+# Copy library files to a lib sub-directory
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${LIB_BINARY_DIR}/lib")
+
+set(CMAKE_THREAD_PREFER_PTHREADS ON)
+set(THREADS_PREFER_PTHREAD_FLAG ON)
+
+set(WITH_ZLIB OFF)
+set(WITH_SYMBOL_VERSIONING OFF)
+set(WITH_SERVER ON)
+
+include(IncludeSources.cmake)
+if (OS_LINUX)
+    if (ARCH_AMD64)
+        if (USE_MUSL)
+            target_include_directories(_ssh PRIVATE "${ClickHouse_SOURCE_DIR}/contrib/libssh-cmake/linux/x86-64-musl")
```

Comment:
How were these files generated (e.g. which make/cmake options)?  Someone is trying to [upgrade](https://github.com/ClickHouse/ClickHouse/pull/59563) libssl from v0.97 to v0.98 and the architecture files need to be re-generated. There is a chance that things will continue to work without regeneration but it will be nicer to have the steps documented somewhere.